satpy/readers/netcdf_utils.py

```diff
@@ -287,3 +321,24 @@ def get(self, item, default=None):
             return self[item]
         else:
             return default
+
+    def get_and_cache_npxr(self, var_name):
+        """Get item as numpy-xarray and keep in cache."""
+        v = self.file_content[var_name]
+        return _cache_npxr(v)
```

Comment:
The problem with this is now it caches on the variable object and not the `var_name` so it isn't saving as much right?

Reply:
If I've understood correctly, if both `var_name` and `v` are passed, both of them are hashed to generate the cache key. So the saving would be even less. The cached method would be the best option, but you asked if we could convert it to a function instead.

Reply:
Ok, converted to use the version in the video.

Reply:
Yeah I think (hope) it is the same amo7nt of hashing but the caching is now per-instance and won't hold on to a reference to self. Hopefully.

Reply:
Blargh. That works well IRL but not in the tests. Need to investigate further tomorrow.

Reply:
The simplest fix (the first) I could think of were to add the cache wrapper to `FakeNetCDF4FileHandler` in the tests. Apparently the `NetCDF4FileHandler` isn't initialized by it.